src/aoscene.cpp

```diff
@@ -1,119 +1,374 @@
+    pathlist = {
+        ao_app->get_image_suffix(ao_app->get_character_path(
+            p_params.at(0), p_params.at(2) + p_params.at(1))), // Default path
+        ao_app->get_image_suffix(ao_app->get_character_path(
+            p_params.at(0),
+            p_params.at(2) + "/" +
+                p_params.at(
+                    1))), // Path check if it's categorized into a folder
+        ao_app->get_image_suffix(ao_app->get_character_path(
+            p_params.at(0),
+            p_params.at(
+                1))), // Just use the non-prefixed image, animated or not
+        ao_app->get_image_suffix(
+            ao_app->get_theme_path("placeholder")), // Theme placeholder path
+        ao_app->get_image_suffix(ao_app->get_default_theme_path(
+            "placeholder"))}; // Default theme placeholder path
+    m_char = p_params.at(0);
+    m_emote = p_params.at(2) + p_params.at(1);
+    break;
+  case image_type::interjection:
+    duration = p_params.at(3).toInt();
+    play_once = true;
+    pathlist = {
+        ao_app->get_image_suffix(ao_app->get_character_path(
+            p_params.at(1), p_params.at(0))), // Character folder
+        ao_app->get_image_suffix(
+            ao_app->get_misc_path(p_params.at(2), p_params.at(0))), // Misc path
+        ao_app->get_image_suffix(
+            ao_app->get_theme_path(p_params.at(0))), // Theme path
+        ao_app->get_image_suffix(ao_app->get_default_theme_path(
+            p_params.at(0))), // Default theme path
+        ao_app->get_image_suffix(
+            ao_app->get_theme_path("placeholder")), // Placeholder path
+        ao_app->get_image_suffix(ao_app->get_default_theme_path(
+            "placeholder")), // Default placeholder path
+    };
+    break;
+  case image_type::effect:
+    effect_duration = p_params.at(3).toInt();
+    is_effect = true;
+    pathlist = {
+        ao_app->get_image_suffix(ao_app->get_character_path(
+            p_params.at(1), "effects/" + p_params.at(0))), // Character folder
+        ao_app->get_image_suffix(ao_app->get_misc_path(
+            p_params.at(2), "effects/" + p_params.at(0))), // Misc path
+        ao_app->get_image_suffix(
+            ao_app->get_theme_path("effects/" + p_params.at(0))), // Theme path
+        ao_app->get_image_suffix(ao_app->get_default_theme_path(
+            "effects/" + p_params.at(0))), // Default theme path
+        ao_app->get_image_suffix(
+            ao_app->get_theme_path("placeholder")), // Placeholder path
+        ao_app->get_image_suffix(ao_app->get_default_theme_path(
+            "placeholder")), // Default placeholder path
+    };
+    break;
+  case image_type::fuck_interface:
+    pathlist = {
+        ao_app->get_image_suffix(
+            ao_app->get_misc_path(p_params.at(1), p_params.at(0))),
+        ao_app->get_image_suffix(ao_app->get_theme_path(p_params.at(0))),
+        ao_app->get_image_suffix(
+            ao_app->get_default_theme_path(p_params.at(0)))};
+    break;
+  }
```

Comment:
Sounds like you should use table lookup instead. You'll have to get a little more inventive though.

Reply:
Any idea where I should start if I wanted to learn table lookup?

Reply:
I recommend a constant `QMap<image_type, QStringList>` and then perhaps using variable expansion to fill in the path to the misc folder, etc, unless I am missing something here (like if some `get_*_path` also has an internal order of precedence).

```cpp
const QMap<image_type, QStringList> precedences = {
 { LayerType::background, { BG_CURRENT, BG_DEFAULT },
 { LayerType::foreground, { MISC_CURRENT, THEME_CURRENT, THEME_DEFAULT },
 // ...
};
```
But this is probably far more complicated and with far less benefit than what you have already written.

Reply:
![image](https://user-images.githubusercontent.com/32779090/93388063-c2964200-f82f-11ea-930d-f2005211dee4.png)
is there a wontfix tag for code reviews